.github/workflows/nightly-ami.yml

```diff
@@ -0,0 +1,96 @@
+name: Nightly AMI Build
+
+on:
+  schedule:
+    # Run every day at 2 AM UTC (adjust timezone as needed)
+    - cron: '0 2 * * *'
+  workflow_dispatch:
+    inputs:
+      boundless_bento_version:
+        description: 'Bento version to build (default: latest)'
+        required: false
+        default: 'latest'
+      boundless_broker_version:
+        description: 'Broker version to build (default: v1.0.0)'
+        required: false
+        default: 'v1.0.0'
+
+concurrency:
+  group: nightly-ami-build
+  cancel-in-progress: false
+
+permissions:
+  id-token: write
+  contents: read
+
+env:
+  AWS_REGION: us-west-2
+  INSTANCE_TYPE: c7a.4xlarge
+  BOUNDLESS_BENTO_VERSION: ${{ github.event.inputs.boundless_bento_version || 'latest' }}
+  BOUNDLESS_BROKER_VERSION: ${{ github.event.inputs.boundless_broker_version || 'v1.0.0' }}
+
+jobs:
+  build-nightly-ami:
+    name: Build Nightly AMI
+    runs-on: [ self-hosted, prod, "Linux", "cpu" ]
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          submodules: recursive
+
+      - name: Configure AWS credentials
+        uses: aws-actions/configure-aws-credentials@v4
+        with:
+          role-to-assume: ${{ secrets.AWS_ROLE_ARN }}
+          aws-region: ${{ env.AWS_REGION }}
```

Comment:
Another reason- we really want to minimize the creds we need in Github actions. That was one of the big motivations for us putting all our deployment infra into AWS rather than GH actions